R/MixedModelsCommon.R

```diff
@@ -15,408 +15,651 @@
-  #             "The argument options should be the options list from QML.",
-  #             "Ensure that the SetSeed{} QML component is present in the QML file for this analysis."))
+  
+  return()
 }
 
 ### common mixed-models functions
-.mmReadData      <- function(dataset, options, type = "LMM"){
-  if(!is.null(dataset)){
+.mmReadData      <- function(dataset, options, type = "LMM") {
+  if (!is.null(dataset)) {
     return(dataset)
-  }else{
-    if(type == "LMM"){
-      return(readDataSetToEnd(columns = c(options$dependentVariable, options$fixedVariables, options$randomVariables)))
-    }else if(type == "GLMM"){
-      if(options$dependentVariableAggregation == ""){
-        return(readDataSetToEnd(columns = c(options$dependentVariable, options$fixedVariables, options$randomVariables)))       
-      }else{
-        return(readDataSetToEnd(columns = c(options$dependentVariable, options$fixedVariables, options$randomVariables, options$dependentVariableAggregation)))
+  } else{
+    if (type %in% c("LMM","BLMM")) {
+      return(readDataSetToEnd(
+        columns = c(
+          options$dependentVariable,
+          options$fixedVariables,
+          options$randomVariables
+        )
+      ))
+    } else if (type %in% c("GLMM","BGLMM")) {
+      if (options$dependentVariableAggregation == "") {
+        return(readDataSetToEnd(
+          columns = c(
+            options$dependentVariable,
+            options$fixedVariables,
+            options$randomVariables
+          )
+        ))
+      } else{
+        return(readDataSetToEnd(
+          columns = c(
+            options$dependentVariable,
+            options$fixedVariables,
+            options$randomVariables,
+            options$dependentVariableAggregation
+          )
+        ))
       }
     }
   }
 }
-.mmCheckData     <- function(dataset){
-  .hasErrors(dataset, type = c('variance', 'infinity'), exitAnalysisIfErrors = TRUE)
+.mmCheckData     <- function(dataset, options, type = "LMM") {
+  .hasErrors(
+    dataset,
+    type = c('variance', 'infinity'),
+    exitAnalysisIfErrors = TRUE
+  )
+  if (type %in% c("GLMM","BGLMM")) {
+    family_text <- .mmMessageGLMMtype(options$family, options$link)
+    family_text <- substr(family_text, 1, nchar(family_text) - 1)
+    if (options$family %in% c("Gamma", "inverse.gaussian", "neg_binomial_2", "poisson")) {
+      if (any(dataset[, .v(options$dependentVariable)] < 0))
+        stop(gettextf("%s requres that the dependent variable is > 0.",family_text))
+    } else if (options$family %in% c("binomial")) {
+      if (any(!dataset[, .v(options$dependentVariable)] %in% c(0, 1)))
+        stop(gettextf("%s requres that the outcomes are only 0 and 1.",family_text))
+    } else if (options$family %in% c("betar")) {
+      if (any(dataset[, .v(options$dependentVariable)] < 0))
+        stop(gettextf("%s requres that the outcomes is higher than 0 and lower than 1.",family_text))
```

Comment:
`JASP:::.quitAnalysis` rather than `stop`.

Reply:
thanks, didn't know about this function before